### DISCUSSION TITLE
My training stopped and I want to continue the training (I had saved on a file but didn't end to train)

### DISCUSSION THREAD

**Discussion by keyssi93** [explosion/spaCy#7842] [Help: Coding & Implementations] [answered]
I had use the basic config from spacy to train with trf 

`!python -m spacy train config.cfg --paths.train ./data/train-loc/ --paths.dev ./data/dev-loc/ --output ./spacy_train_location_8 --gpu-id 0
`
But it has stoped 
<img width="381" alt="Capture2" src="https://user-images.githubusercontent.com/49962161/115450319-5c1c3c80-a1e1-11eb-8a4d-473ff3e4b584.PNG">

I would like to know how to continue training that model to get better score 

## Your Environment
- Operating System:
- Python Version Used:Python 3.7.10
- spaCy Version Used:spacy v3.0
- Environment Information:pytorch_latest_p37

**Comment by polm** [comment, marked as answer]
If you want to train the model for longer, you should adjust the max epochs, max steps, or patience setting. See [here](https://github.com/explosion/spaCy/discussions/7465) for an explanation of how those interact. 

You can also use what you've trained so far as a starting point, just set `source = my_model` under the component you're training; you can see that [here](https://spacy.io/usage/training#config-components:~:text=.%20If%20a%20component%20block%20defines,trained%20pipeline%2C%20with%20its%20existing%20weights.) in the docs.

**Reply by keyssi93** [reply]
Hi thanks for the reply, But what I don't uderstand is that I stored my model in folder and I want to use that so you should I change "my_model" for the path of the folder that contain it ?

![image](https://user-images.githubusercontent.com/49962161/115599083-c516b980-a2a0-11eb-876d-9df3345e68dd.png)

![image](https://user-images.githubusercontent.com/49962161/115599129-d8c22000-a2a0-11eb-8d50-28dbe9a5ecec.png)

**Reply by polm** [reply]
`source` should be the name of the folder with your model, yes. It should not be the json file. 

Note that you need to remove the "factory" entry when you supply a "source" entry.

**Reply by adrianeboyd** [reply]
To continue training a transformer model, be sure to source the `transformer` and all the other components:

```ini
[components.transformer]
source = "/path/to/model-best"

[components.tagger]
source = "/path/to/model-best"

[components.ner]
source = "/path/to/model-best"
```

You can make a copy of the config in `model-best/config.cfg` and just replace all the component config sections with `source = `.
